Source/Noesis.Javascript/JavascriptContext.cpp

```diff
@@ -265,12 +275,27 @@ JavascriptContext::SetParameter(System::String^ iName, System::Object^ iObject,
 		}
 	}
 
-	v8::Local<v8::String> key = String::NewFromTwoByte(isolate, (uint16_t*)name, v8::NewStringType::kNormal).ToLocalChecked();
+    v8::Local<v8::String> key = ToV8String(isolate, iName);
 	Local<Context>::New(isolate, *mContext)->Global()->Set(isolate->GetCurrentContext(), key, value).ToChecked();
 }
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////
 
+
+void JavascriptContext::SetConstructor(System::String^ name, System::Type^ associatedType, System::Delegate^ constructor)

```

Comment:
This would be nicer written as a generic (in C# SetConstructor<T>(string name, Func<T> constructor).  How is this done in C++?  It might be something like

generic <class T>void SetConstructor<T>(System::String ^name, System::Func<T> ^constructor)

This is just a suggestion to make the API nicer to use; perhaps it is more trouble than it is worth.

Reply:
Hm, yeah. I guess this makes sense for people not using reflection for calling this :P (so probably everybody except us 😆). I will add a second function which accepts the type via generics (which simply calls this one) if that is ok?

Reply:
Thanks.